app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -449,12 +477,46 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent)
             format = result.getFormatName();
         }
 
-        if(requestCode == SELECT_BARCODE_REQUEST && resultCode == Activity.RESULT_OK)
+        if (resultCode == Activity.RESULT_OK)
         {
-            Log.i(TAG, "Received barcode information from typing it");
+            if(requestCode == SELECT_BARCODE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode information from typing it");
 
-            contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
-            format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+                contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
+                format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+            }
+            else if (requestCode == SELECT_IMAGE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode image");
+
+                Bitmap bitmap = null;
+                try {
+                    bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), intent.getData());
+                } catch (IOException e) {
+                    Log.e(TAG, "Error getting the image data");
+                    e.printStackTrace();
```

Comment:
Same here, should this be logged instead?

Reply:
Right, printStacktrace() is, in fact, logged, but without the TAG. I'll move it into the Log.e call.